sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1801,7 +1801,7 @@ BRACKETED_EMPTY_COMMENT
     ;
 
 BRACKETED_COMMENT
-    : '/*' ~[+] .*? '*/' -> channel(HIDDEN)
+    : '/*' ~[+] (BRACKETED_COMMENT|.)*? '*/' -> channel(HIDDEN)
```

Comment:
It will fail
```
/*/**/*/
```

Reply:
Actually does anyone know why there is `~[+]` here?

Reply:
For hints?

Reply:
@maropu ah, right

Reply:
Yea!

Reply:
I will take a look.

Reply:
can we create a lexer rule for hint and put it before this one?

Reply:
@gengliangwang Now, it works well.

Reply:
@cloud-fan I fixed the issue @gengliangwang proposed.

Reply:
How about the @cloud-fan comment? The fix looks nice if we can. Its okay to fix it in followup though.

Reply:
@beliefer my point of the case 
```
/*/**/*/
```
is not about empty comment. It is about another `/*` after `/*`.
We can change the case to 
```
/*/*foo*/*/
```
The latest fix will still failed.